fluss-server/src/test/java/com/alibaba/fluss/server/kv/rocksdb/RocksDBMetricsManagerTest.java

```diff
@@ -0,0 +1,423 @@
+            Thread.currentThread().interrupt();
+        }
+
+        // Normal collector should still be called despite exception in faulty collector
+        verify(spyNormalCollector, times(1)).updateMetrics();
+    }
+
+    @Test
+    void testShutdown() {
+        RocksDBMetricsManager manager = RocksDBMetricsManager.getInstance();
+
+        // Register some collectors
+        for (int i = 0; i < 3; i++) {
+            RocksDBMetricsCollector collector = createMockCollector("test_db", "test_table", i);
+            manager.registerCollector(collector);
+            collectors.add(collector);
+        }
+
+        assertThat(manager.isRunning()).isTrue();
+        assertThat(manager.getCollectorCount()).isEqualTo(3);
+
+        manager.shutdown();
+
+        assertThat(manager.isRunning()).isFalse();
+        assertThat(manager.isShutdown()).isTrue();
+        assertThat(manager.getCollectorCount()).isEqualTo(0);
+    }
+
+    @Test
+    void testDoubleShutdown() {
+        RocksDBMetricsManager manager = RocksDBMetricsManager.getInstance();
+
+        manager.shutdown();
+        assertThat(manager.isShutdown()).isTrue();
+
+        // Second shutdown should not cause issues
+        manager.shutdown();
+        assertThat(manager.isShutdown()).isTrue();
+    }
+
+    @Test
+    void testConcurrentRegistrationAndUnregistration() throws Exception {
+        RocksDBMetricsManager manager = RocksDBMetricsManager.getInstance();
+
+        int threadCount = 10;
+        CountDownLatch latch = new CountDownLatch(threadCount);
+        ExecutorService executor = Executors.newFixedThreadPool(threadCount);
+        AtomicInteger collectorId = new AtomicInteger(0);
+
+        for (int i = 0; i < threadCount; i++) {
+            executor.submit(
+                    () -> {
+                        try {
+                            for (int j = 0; j < 10; j++) {
+                                RocksDBMetricsCollector collector =
+                                        createMockCollector(
+                                                "test_db",
+                                                "test_table",
+                                                collectorId.getAndIncrement());
+                                manager.registerCollector(collector);
+                                collectors.add(collector);
+
+                                // Immediately unregister some collectors
+                                if (j % 2 == 0) {
+                                    manager.unregisterCollector(collector);
+                                }
+                            }
+                        } finally {
+                            latch.countDown();
+                        }
+                    });
+        }
+
+        latch.await(10, TimeUnit.SECONDS);
+        executor.shutdown();
+
+        // Should handle concurrent access gracefully
+        assertThat(manager.getCollectorCount()).isGreaterThanOrEqualTo(0);
```

Comment:
I think we can assert the count value `50` here? It should be deterministic, right?